ReactWindows/ReactNative.Net46.Tests/Modules/Storage/AsyncStorageModuleTests.cs

```diff
@@ -37,14 +38,18 @@ public void Init()
         }
 
         [TearDown]
-        public void Cleanup()
+        public void TearDown()
         {
-            module.clear(callback);
-            Assert.That(waitHandle.WaitOne(), Is.True);
-            Assert.That(error, Is.Null);
-            Assert.That(result, Is.Null);
+            module.clear(new MockCallback((invoke) => { }));
         }
 
+        [Test]
+        public void AsyncStorageModule_DoesNotInvokeCallbackAfterDispose()
+        {
+            callback = new MockCallback(invoke => throw new Exception("should not be called"));
```

Comment:
To make this test a bit more deterministic, can we use a ManualResetEvent and call Wait with a timeout (say 500ms)? Otherwise, if this regresses, the exception that gets thrown could become a red herring / cause another test to fail / cause the test runner to crash.